plugin-cgroupv2/src/k8s/token.rs

```diff
@@ -0,0 +1,140 @@
+use std::{
+    fs::File,
+    io::Read,
+    process::Command,
+    sync::Arc,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use anyhow::anyhow;
+use base64::{prelude::BASE64_STANDARD_NO_PAD, Engine};
+use tokio::sync::RwLock;
+
+use super::plugin::TokenRetrieval;
+
+pub struct Token {
+    expiration_time: RwLock<Option<u64>>,
+    token_retrieval: TokenRetrieval,
+    value: RwLock<Option<String>>,
```

Comment:
It's dangerous to use two separate `RwLock`: while you modify the token, another thread may see a partially-updated struct (e.g. expiration time of token A with value of token B).

You can also make the token struct `Send` by using `Arc` directly, this will make it fully "thread-safe" and simplify its use in the plugin. It would also simplify its implementation :)
```rust
pub struct Token {
    retrieval: TokenRetrieval, // could also be in TokenData, it doesn't change much here
    data: Arc<RwLock<TokenData>>
}

/// Private structure that holds the JWT token data.
struct TokenData {
    /// Optional expiration time, in POSIX seconds.
    expiration_time: Option<u64>,
    /// Before the first retrieval, there is no token.
    value: Option<String>,
}
```

Reply:
Note: with this `Arc<RwLock<TokenData>>`, you should use `self` instead of `self: Arc<Self>` in the Token impl